### PR TITLE
[JS] Others : Data Types in JavaScript

### LINKED ISSUE #73: JS: Data Types in JavaScript
# Data Types in JavaScript
Discuss the built-in data types in JavaScript.

## Directory
```
Topics/5.Syntax/1.Data_Types
```

## Description
- Talk about the various data types in JavaScript.
- String, boolean, number, null, undefined, object.
- What is the `BigInt` type?
- How do we know the type of a particular value held in a variable?
  Hint: Using `typeof`.

## Hints
- Here's a lovely article to get you started: [Data Types in JavaScript](https://javascript.info/types).
- Check out the [W3C tutorial](https://www.w3schools.com/js/).

### PR DESCRIPTION
**Pull request by AbhinandanAdhikari**
<hr>

## Description 📜

I've updated the already made README file for JavaScript Data Types with some additional contents.
Fixes #73

<hr>

## Type of change 📝

- [ ] Audio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Video (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev (Flutter, Java, Kotlin)
- [ ] Backend Dev (Java, .NET, Python)
- [ ] C/CPP
- [ ] Competitive Programming
- [ ] Cyber Security
- [ ] DSA
- [ ] Database
- [ ] Datascience with Python
- [ ] Datascience with R
- [ ] Frontend Dev HTML/CSS/JS
- [ ] Frontend Dev React/Angular/Vue
- [ ] Golang
- [ ] Interview Prep
- [ ] Java_Domain
- [x] JavaScript
- [ ] MERN
- [ ] Machine Learning
- [ ] Open Source
- [ ] Python_Domain
- [ ] Rust
- [ ] Statistics
- [ ] UI/UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by AbhinandanAdhikari** [comment]
@paramsiddharth  Sir, as told by you I've updated the already merged PR by Aniket with some additional contents, Please review this. And please remove the other domain labels i don't know how it got added.

**Comment by AbhinandanAdhikari** [comment]
> @paramsiddharth Sir, as told by you I've updated the already merged PR by Aniket with some additional contents, Please review this. And please remove the other domain labels i don't know how it got added.

@dakshsethi @Creatoon @s4-2 Mentors! Please review this.

**Review by s4-2 — APPROVED** [review]
Looks good to me:)

**Review by dakshsethi — APPROVED** [review]
Looks good to me!! 🔥🔥🔥

File: JavaScript/Topics/5.Syntax/1.Data_Types/README.md [hunk 1]
```diff
@@ -100,4 +106,29 @@ let person = {
     age:50, 
     eyeColor:"blue"};
 ```
+## 2. Arrays
+Arrays are collection of values of similar data type.
```
Comment (Creatoon) [inline JavaScript/Topics/5.Syntax/1.Data_Types/README.md:110]: In javascript, arrays can be a collection of different data types

**Review by paramsiddharth — APPROVED** [review]

File: JavaScript/Topics/5.Syntax/1.Data_Types/README.md (hunk 1 above)
Comment (AbhinandanAdhikari) [reply JavaScript/Topics/5.Syntax/1.Data_Types/README.md]: Sir, I've made the changes now review it.

**Comment by AbhinandanAdhikari** [comment]
@paramsiddharth Sir, now review

File: JavaScript/Topics/5.Syntax/1.Data_Types/README.md [hunk 2]
```diff
@@ -100,4 +106,29 @@ let person = {
     age:50, 
     eyeColor:"blue"};
 ```
+## 2. Arrays
+Arrays are collection of values of similar data type. In JavaScript, array can also be a collection of data of different data types.
```
Comment (Creatoon) [inline JavaScript/Topics/5.Syntax/1.Data_Types/README.md:110]:
```suggestion
Arrays are collection of values of similar or different data types.
```

File: JavaScript/Topics/5.Syntax/1.Data_Types/README.md [hunk 3]
```diff
@@ -100,4 +106,29 @@ let person = {
...
+
+JavaScript Arrays are written with square bracket<kbd>[ ]</kbd>, seperated by commas.
+
+__Example:__
+```javascript
+const bike = ["BMW","Honda","Suzuki"];
+```
+Array indexes are zero-based, means the first item is at [0].
```
Comment (Creatoon) [inline JavaScript/Topics/5.Syntax/1.Data_Types/README.md:118]:
```suggestion
Array indexes are zero-based, means the first item is at  arrayName[0].
```

**Review by Creatoon — CHANGES_REQUESTED** [review]

**Comment by AbhinandanAdhikari** [comment]
@paramsiddharth @Creatoon  Sir, made the changes now review it.

**Review by Creatoon — APPROVED** [review]

**Comment by Creatoon** [comment]
LGTM

**Pull request merged**